arch/src/x86_64/cpu_profile_feature_flags.rs

```diff
@@ -14,74 +14,74 @@ pub(super) struct CascadeLakeServerV1CpuIdFeatures {
 }
 
 impl CascadeLakeServerV1CpuIdFeatures {
-    pub(super) fn new() -> Self {
+    pub(super) const fn new() -> Self {
         use CpuIdFeatureFlags as FF;
-        Self {
-            edx_1: FF::VME
-                | FF::SSE2
-                | FF::SSE
-                | FF::FXSR
-                | FF::MMX
-                | FF::CLFLUSH
-                | FF::PSE36
-                | FF::PAT
-                | FF::CMOV
-                | FF::MCA
-                | FF::PGE
-                | FF::MTRR
-                | FF::SEP
-                | FF::APIC
-                | FF::CX8
-                | FF::MCE
-                | FF::PAE
-                | FF::MSR
-                | FF::TSC
-                | FF::PSE
-                | FF::DE
-                | FF::FPU,
+        // Placing this in a const block ensures compile time evaluation and we get isntant feedback
+        // (even from the LSP) if the lists contain arguments that are not defined for the given
+        // function, index, register triple via `crate::x86_64::impl_cpuid_feature_flags!`
+        const {
+            Self {
+                edx_1: FF::<1, 0, { CpuidReg::EDX as u8 }>::from_names(&[
```

Comment:
For better or worse: The compiler does not manage to infer the const generic parameters when they are left out here, even though we have specified them in the field declaration of the struct.